components/collector/tests/source_collectors/github/test_merge_requests.py

```diff
@@ -0,0 +1,181 @@
+"""Unit tests for the GitHub merge requests collector."""
+# Author: Tobias Termeczky
+# Company: the/experts.
+
+from unittest.mock import AsyncMock, patch
+
+import aiohttp
+
+from base_collectors import MetricCollector
+
+from .base import GitHubTestCase
+
+
+class GitHubMergeRequestsTest(GitHubTestCase):
+    """Unit tests for the merge request metric."""
+
+    METRIC_TYPE = "merge_requests"
+    LANDING_URL = "https://github/owner/repository/pulls"
+
+    @staticmethod
+    def merge_request_json(
+        number: int,
+        branch: str = "default",
+        state: str = "MERGED",
+        review_decision: str | None = None,
+    ) -> dict[str, int | str | dict[str, int] | None]:
+        """Create a merge request."""
+        return {
+            "id": number,
+            "title": f"Merge request {number}",
+            "baseRefName": branch,
+            "state": state,
+            "url": f"https://github/pull{number}",
+            "createdAt": "2017-04-29T08:46:00Z",
+            "updatedAt": "2017-04-29T09:40:00Z",
+            "mergedAt": None,
+            "reviewDecision": review_decision,
+            "reactions": {"totalCount": number},
+            "comments": {"totalCount": number},
+        }
+
+    @staticmethod
+    def merge_requests_json(
+        nodes,
+        count: int | None = None,
+        has_next_page: bool = False,
+    ) -> dict[str, dict[str, dict[str, dict[str, dict[str, str | bool] | int]]]]:
```

Comment:
This could also make use of the `GitHubMergeRequestJSON` type introduced above, I suspect.

Reply:
No, these 2 are different, make this a TypedDict wil result in the creation of many classes.